### PR TITLE
Calculate iou per class

### PR DESCRIPTION
**Pull request by hdefazio**
Initial version of the evaluation suite. 

Metrics:
IoU

Plots:
Activity detections (optimal from @ Purg 's demo notebook and real detections)

### CODE REVIEW

**Comment by hdefazio** [comment]
@alexander-lynch Could you double check compute_scores.py?

**Comment by alexander-lynch** [comment]
compute_scores.py looks good to me

**Comment by hdefazio** [comment]
@Purg / @alexander-lynch  PTAL

File: pyproject.toml [hunk 1]
```diff
@@ -19,6 +19,8 @@ transitions = "^0.8.11"
 timm = "^0.5.4"
 einops = "^0.4.1"
 pynput = "^1.7.6"
+pandas = "^1.4.3"
+pyarrow = "^9.0.0"
```
[the remark this answers is not in the document]
Comment (hdefazio) [reply pyproject.toml:23]: pandas is being used to read in the annotation feather file, I think I can remove pyarrow though

File: angel_system/eval/visualization.py [hunk 2]
```diff
@@ -19,17 +19,20 @@ def plot_activity_confidence(label, gt_ranges, det_ranges, output_dir, custom_ra
...
-        ax.set_ylim(0, 1.05)
-        ax.set_xlim(GlobalValues.slice_time_ranges[0][0] - 1,
-                    GlobalValues.slice_time_ranges[-1][1] + 1)
-        # plt.yscale("log")
+    all_start_times = [p["time"][0] for p in gt_ranges]
+    all_start_times.extend([p["time"][0] for p in det_ranges[label]])
+    all_end_times = [p["time"][1] for p in gt_ranges]
+    all_end_times.extend([p["time"][1] for p in det_ranges[label]])
+    min_start_time = min(all_start_times) - 1
+    max_end_time = max(all_end_times) + 1
```
Comment (Purg) [inline angel_system/eval/visualization.py:27]: The +-1 part was somewhat hard-coded values from the example notebooks. Using something that is percentile based would be more stable for different length scales of the input data (e.g. pad 5% total time delta to the beginning and end of the plot).

File: angel_system/eval/visualization.py [hunk 3]
```diff
@@ -84,14 +62,15 @@ def plot_activity_confidence(label, gt_ranges, det_ranges, output_dir, custom_ra
...
+    ax.legend(loc="upper right")
+    ax.plot
 
     # ============================
     # Save
     # ============================
     #plt.show()
     Path(os.path.join(output_dir, "plots/activities")).mkdir(parents=True, exist_ok=True)
     fig.savefig(f"{output_dir}/plots/activities/{label.replace(' ', '_')}.png")
+
```
Comment (Purg) [inline angel_system/eval/visualization.py:76]: Extraneous new-line.

**Review by Purg — COMMENTED** [review]
With the code removed from `visualization.py`, I think it's now valuable to consider removing the `GlobalValues` and `SelectedSlice` structures: I cannot find their legitimate usage now.

File: angel_system/eval/compute_scores.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            else:
+                # gt starts before detection, ignore part of gt that happens before our detection range
+                intersection_start = det_range["time"][0]
+
+            if det_range["time"][0] <= gt_overlap[1] <= det_range["time"][1]:
+                # gt ends before detection
+                intersection_end = gt_overlap[1]
+            else:
+                # detection ends before gt, ignore part of gt that happens after the detection range
+                intersection_end = det_range["time"][1]
```
Comment (Purg) [inline angel_system/eval/compute_scores.py:57]: I believe the intersection area code could be simpler, unless I'm forgetting something, e.g. see [this](https://stackoverflow.com/a/41660682/1128142) which is for boxes, but remove the y dimension.
```python
# coordinates of the intersection interval
i_left = max(det_range['time'][0], gt_range['time'][0])  # "right"-most left boundary
i_right = min(det_range['time'][1], gt_range['time'][1])  # "left"-most right boundary
intersection_area = i_right - i_left
```

**Review by Purg — COMMENTED** [review]
I think this is basically fine for now (last comment I swear). We will spin out specific paths forward from here in the meet-up we have scheduled.

**Review by Purg — APPROVED** [review]

**Pull request merged**